File: src/main/java/net/lomeli/lomlib/recipes/anvil/IAnvilRecipe.java
package net.lomeli.lomlib.recipes.anvil;

import net.minecraft.item.ItemStack;

public interface IAnvilRecipe {
    /**
     * Checks to see if the left and right stack match the recipe
     * @param left
     * @param right
     * @return
     */
    boolean matches(ItemStack left, ItemStack right);
    
    /**
     * Get a copy of the result
     * @param left
     * @param right
     * @return
     */
    ItemStack getCraftingResult(ItemStack left, ItemStack right);
    
    /**
     * Get the recipe result
     * @return
     */
    ItemStack recipeOutput();
    
    /**
     * Get the inputs for the recipe
     * @return
     */
    Object[] recipeInputs();
    
    /**
     * Cost (in Experience levels) of the recipe
     * @return
     */
    int recipeCost();
}

File: src/main/kotlin/net/lomeli/lomlib/core/recipes/IAnvilRecipe.kt
package net.lomeli.lomlib.core.recipes

import net.minecraft.item.ItemStack

interface IAnvilRecipe {
    /**
     * Checks to see if the left and right stack match the recipe.

     * @param left
     * *
     * @param right
     * *
     * @return
     */
    fun matches(left: ItemStack, right: ItemStack?): Boolean

    /**
     * Get a copy of the result

     * @param left
     * *
     * @param right
     * *
     * @return
     */
    fun getCraftingResult(left: ItemStack, right: ItemStack?): ItemStack

    /**
     * Get the recipe result

     * @return
     */
    fun recipeOutput(): ItemStack

    /**
     * Get the inputs for the recipe

     * @return
     */
    fun recipeInputs(): Array<Any?>

    /**
     * Cost (in Experience levels) of the recipe

     * @return
     */
    fun recipeCost(): Int

    fun doItemsMatch(itemStack: ItemStack?, slot: Int): Boolean
}